pkg/deploy/controller/deploymentcancellation/controller.go

```diff
@@ -0,0 +1,77 @@
+package deploymentcancellation
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/record"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+)
+
+// DeploymentCancellationController monitors for deployments that have been
+// cancelled by the user and sets the ActiveDeadlineSeconds to 0 on the
+// corresponding deployer pods.
+//
+// Use the DeploymentCancellationControllerFactory to create this controller.
+type DeploymentCancellationController struct {
+	// podClient provides access to pods.
+	podClient podClient
+	recorder  record.EventRecorder
+}
+
+// Handle processes deployment and either creates a deployer pod or responds
+// to a terminal deployment status.
+func (c *DeploymentCancellationController) Handle(deployment *kapi.ReplicationController) error {
+	currentStatus := deployutil.DeploymentStatusFor(deployment)
+
+	switch currentStatus {
+	case deployapi.DeploymentStatusNew,
+		deployapi.DeploymentStatusPending,
+		deployapi.DeploymentStatusRunning:
+
+		if !deployutil.IsDeploymentCancelled(deployment) {
+			return nil
+		}
+
+		deployerPod, err := c.podClient.getPod(deployment.Namespace, deployutil.DeployerPodNameFor(deployment))
```

Comment:
That is correct. The cancellation annotation is just to capture user intent. Based on that, we set the deployer pod ActiveDeadlineSeconds to 0. This results in the pod containers being killed and pod getting to a Failed state. This pod state/phase change triggers the state of the deployment being updated to Failed.